src/precomputed/index.jl

```diff
@@ -0,0 +1,185 @@
+                        not_available[n] = result
+                        []
+                    else
+                        result
+                    end
+                end for n in names
+            ]
+
+            VariableGroupPossibilities(;
+                names=names,
+                possible_values=possible_values,
+                not_available=not_available,
+                num_possibilities=prod(BigInt.(length.(possible_values))),
+            )
+        end for variable_group in filter(!isempty, Set(values(connections)))
+    ]
+end
+
+
+function combination_iterator(group::VariableGroupPossibilities)
+    Iterators.map(Iterators.product(group.possible_values...)) do combination
+        bonds_dict = OrderedDict{Symbol,Any}(
+            n => OrderedDict{String,Any}("value" => v) for
+            (n, v) in zip(group.names, combination)
+        )
+
+        return (combination, bonds_dict)
+    end
+end
+
+biglength(pr::Iterators.ProductIterator) = prod(BigInt[biglength(i) for i in pr.iterators])
+biglength(g::Base.Generator) = biglength(g.iter)
+biglength(x) = BigInt(length(x))
+
+function generate_precomputed_staterequests_report(
+    groups::Vector{VariableGroupPossibilities},
+    run::RunningNotebook;
+    settings::PlutoDeploySettings,
+    pluto_session::Pluto.ServerSession,
+)::PrecomputedSampleReport
+    map(groups) do group
+        stat = if !isempty(group.not_available)
+            Normal(0.0, 0.0)
+        else
+            iterator = combination_iterator(group)
+            if isempty(iterator) || isempty(group.names)
+                Normal(0.0, 0.0)
+            else
+                file_size_sample =
+                    map(
+                        rand(iterator, length(group.names) * 3),
+                    ) do (combination, bonds_dict)
+
+                        result = run_bonds_get_patches(pluto_session, run, bonds_dict)
+
+                        if result !== nothing
+                            length(Pluto.pack(result))
+                        else
+                            0
+                        end |> BigInt
+                    end .* biglength(iterator) # multiply by number of combinations to get an estimate of the total file size
+
+                fit(Normal, file_size_sample)
+            end
+        end
+        VariableGroupPossibilities(;
+            names=group.names,
+            possible_values=group.possible_values,
+            num_possibilities=group.num_possibilities,
+            not_available=group.not_available,
+            file_size_sample_distribution=stat,
+            settings,
+        )
+    end |> PrecomputedSampleReport
+end
+
+
+function generate_precomputed_staterequests(
+    notebook_session::NotebookSession;
+    settings::PlutoDeploySettings,
+    pluto_session::Pluto.ServerSession,
+    output_dir=".",
+)
+
+    sesh = notebook_session
+    run = sesh.run
+    connections = run.bond_connections
```

Comment:
this causes problems if I have some notebooks under `PlutoSliderServer_exclude`, because in that case `run` will be `FinishedNotebook`.

Myabe one dirty fix could be to have before this

```julia
run isa FinishedNotebook && return
```

?

cleaner might be not to call this in the first place for notebooks under `PlutoSliderServer_exclude`.